webdriver-spec.html

```diff
@@ -2259,23 +2296,55 @@ <h2 id=element-displayedness>Element Displayedness</h2>
    </ol>
   </section>
 </section>
+
 <section>
-  <h2>Element State</h2>
-    <section>
-      <h2>Reading current element state from the document</h2>
-      <p class="note">Conformance tests for this section can be found in the <a href="https://github.com/w3c/web-platform-tests/tree/master/webdriver/">webdriver module</a> under the "element_state" folder.</p>
-
-          <p>When we are returning an <a>ElementRect</a> to the local end it will need to be serialised</p>
-          <dl class='idl' title='dictionary ElementRect'>
-            <dt>double x</dt>
-            <dd>The x coordinate for the top left of the element</dd>
-            <dt>double y</dt>
-            <dd>The y coordinate for the top left of the element</dd>
-            <dt>double height</dt>
-            <dd>Height of the element in <a href='http://www.w3.org/TR/css3-values/#absolute-lengths'>CSS reference pixels</a></dd>
-            <dt>double width</dt>
-            <dd>Width of the element in <a href='http://www.w3.org/TR/css3-values/#absolute-lengths'>CSS reference pixels</a></dd>
-          </dl>
+<h2>Element State</h2>
+
+<p>To <dfn>calculate the absolute position</dfn> of
+ an <a href=https://dom.spec.whatwg.org/#concept-element>element</a>:
+
+<ol>
+ <li><p>Let <var>x</var> be 0.
+
+ <li><p>Let <var>y</var> be 0.
+
+ <li><p>While <var>element</var>’s
+  <a href=http://dev.w3.org/csswg/cssom-view/#dom-htmlelement-offsetparent>offsetParent</a>
+  is not null:
+
+  <ol>
+   <li><p>Set <var>x</var> to (<var>x</var> +
+    <var>element</var>’s <a href=http://dev.w3.org/csswg/cssom-view/#dom-htmlelement-offsetleft>offsetLeft</a>).
+
+   <li><p>Set <var>y</var> to (<var>y</var> +
+    <var>element</var>’s <a href=http://dev.w3.org/csswg/cssom-view/#dom-htmlelement-offsettop>offsetTop</a>).
+
+   <li><p>Set <var>element</var> to <var>element</var>’s
```

Comment:
It’s being used for iteration.  To simplify this with some pseudo code:

```
while (element.offsetParent !== null) {
  x += element.offsetLeft
  y += element.offsetTop
  element = element.offsetParent
```

Eventually `element`’s `offsetParent` will be `null` when it reaches the final element.